python/pyspark/testing/pandasutils.py

```diff
@@ -450,6 +477,15 @@ def assert_eq(self, left, right, check_exact=True, almost=False):
         import pandas as pd
         from pandas.api.types import is_list_like
 
+        # for pandas-on-Spark, ignore row order by default because row ordering not guaranteed
+        if isinstance(left, (ps.DataFrame, ps.Series, ps.Index)):
+            return assertPandasOnSparkEqual(
+                left, right, checkExact=check_exact, almost=almost, checkRowOrder=check_row_order
```

Comment:
Should the default value for check_row_order be false? since we want o ignore the row ordering here.

Reply:
updated the above comment (allowing check_row_order=False for pandas-on-Spark only)